infrastructure/lambda/template.yaml

```diff
@@ -1354,7 +1347,6 @@ Resources:
     Type: AWS::Serverless::Function
     Condition: IsDevEnvironment
     Properties:
-      FunctionName: !Sub "${AWS::StackName}-CreateAuthCodeFunction"
       CodeUri: ../../lambdas
       Handler: create-auth-code-handler.lambdaHandler
       Runtime: nodejs18.x
```

Comment:
Typically we should let aws create the names but there is a reason why this was the exception, all CRI's have a [private-api yaml](https://github.com/govuk-one-login/ipv-cri-kbv-api/blob/main/infrastructure/lambda/private-api.yaml#L36) referencing these names directly. Or have you figured out an alternate way this can be done?

Reply:
Thanks, Ola, this is really useful.

We should be using outputs from the common stack to get the names and ARNs of functions instead of fixed names because 1) it's a special knowledge you need to have in order to not break things (and I wasn't aware of this) and 2) it places an arbitrary constraint on the length of the stack name.

The correct solution is to change all CRIs to use the outputs but that's a bigger piece of work so for now I'll introduce a length limit on the stack name which has to be `64 - (<longest fixed lambda name> + 1)` - 64 is the maximum lambda name length, plus one character for the hyphen. This is really terrible but for now we have to do this.

Reply:
@CharlesIC The maximum is actually a lot less than in many cases, sometimes it's as little as 23 characters!

@mrdoodles knows more about that as he's had battles with infrastructure before

Reply:
I have come across this and in the past I had to set the limit to 20 something chars but in this case the limit is imposed by our own templates - setting it to 37 is enough to deploy a preview stack and run integration tests as we don't use secure pipelines for this - in cases where we do use the pipelines the limit might indeed be even less but that's a different workflow not related to this PR, thankfully.

Reply:
We should add comments to the template above those `FunctionName` lines to say they should not be changed or removed, since this is not obvious without the knowledge of all the CRIs.

Reply:
I've raised https://github.com/govuk-one-login/github-actions/pull/11 and pushed [this commit](https://github.com/govuk-one-login/ipv-cri-common-lambdas/pull/263/commits/b749db683c83d6bd6d3694bafd109e9a71de3d04) to set the lmit for integration test stacks.